src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,6 +17,133 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Removes all leading and trailing white-space characters from the span.
+        /// </summary>
+        /// <param name="span">The span</param>
+        public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span)
```

Comment:
These methods are not generic. I do not think you are gaining anything by spitting them into multiple parts - the implementation should be right here.

Reply:
Is it OK to still re-use TrimStart()/TrimEnd() within the Trim() implementation rather than duplicating the code? Are you suggesting I just inline the calls to the SpanHelpers and remove SpanHelpers.char.cs?

Reply:
Yes, implementing Trim as TrimStart + TrimEnd is fine.